middle_end/flambda2/to_cmm/to_cmm_env.ml

```diff
@@ -783,6 +802,80 @@ let inline_variable ?consider_inlining_effectful_expressions env res var =
           let env = remove_binding env var in
           will_inline_simple env res binding)))
 
+(* Handling of aliases between variables *)
+
+let force_binding_to_be_split t res var =
+  match Variable.Map.find var t.bindings with
+  | exception Not_found -> t, res, None
+  | binding -> (
+    match binding with
+    | Binding
+        { bound_expr = Simple _;
+          inline = (Do_not_inline | May_inline_once) as inline;
+          _
+        } ->
+      t, res, Some (Any_inline inline)
+    | Binding
+        ({ bound_expr = Split _ | Splittable_prim _;
+           inline = (Must_inline_once | Must_inline_and_duplicate) as inline;
+           _
+         } as binding) ->
+      let t, res, _binding = split_in_env t res var binding in
+      t, res, Some (Any_inline inline))
+
+let add_alias t res ~var
+    ~(num_normal_occurrences_of_var : Num_occurrences.t Variable.Map.t)
+    ~alias_of =
+  if debug ()
+  then
+    Format.eprintf "add_alias var=%a (occs %a), alias_of=%a\n%!" Variable.print
+      var
+      (Variable.Map.print Num_occurrences.print)
+      num_normal_occurrences_of_var Variable.print alias_of;
+  let t, res, inline = force_binding_to_be_split t res alias_of in
```

Comment:
In the case of a `Must_inline_once` binding, and when the number of occurrences of the new variable is exactly one, we'd prefer to avoid splitting if it's possible.